#2 Detect if comment or not

shreyanshanchlia opened the pull request:
Write a program to identify whether a given line is a comment or not.

shreyanshanchlia committed:
#2 Detect if comment or not

Write a program to identify whether a given line is a comment or not.

Kishorsarswat commented on is comment or not/comment or not_0.cpp:
```diff
@@ -0,0 +1,100 @@
+    string comment;
+
+    //determines prev char to determine next char
+    enum CommentType
```
excellent approach

Kishorsarswat commented on is comment or not/comment or not_0.cpp:
```diff
@@ -0,0 +1,100 @@
+            {
+                comment += *ch;
+                
+                ch++;
```
This right here breaks the flow. everything checked char by char but here it is incrementing ch. 
you could have added an enum like "multilineEnd" or something to coverup.

Kishorsarswat replied:
As i suspeected, it is resulting as a bug. you can not close a multiline comment with **/

shreyanshanchlia replied:
Fixed

Kishorsarswat committed:
#5 valid operators

This function checks if the current token is a operator or not. this can be checked by an amazing observation about operators. operators are mostly of length 1 or 2. lets consider others as the special case.

Kishorsarswat committed:
Delete is valid operator.cpp

needed

shreyanshanchlia committed:
Multi-line comment fixes  #2

Added additional enum property for multi-line end '*' before '/'

Kishorsarswat reviewed (approved):
fixes are done!
